src/bridge/cockpitcpusamples.c

```diff
@@ -112,3 +112,76 @@ cockpit_cpu_samples (CockpitSamples *samples)
   g_strfreev (lines);
   g_free (contents);
 }
+
+gchar*
+read_hwmon_file (gchar *path, GError **error)
+{
+  gchar *content = NULL;
+
+  if (!g_file_get_contents (path, &content, NULL, error))
+    {
+      if (!g_error_matches (*error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
+          g_message ("error loading name: %s", (*error)->message);
+    }
+
+  return content;
+}
+
+void
+sample_cpu_sensors (CockpitSamples *samples, int hwmonID)
+{
+  for (int i = 1; TRUE; i++)
+    {
+      g_autoptr(GError) error = NULL;
+      g_autofree gchar *temp = g_strdup_printf ("/sys/class/hwmon/hwmon%d/temp%d_input", hwmonID, i);
+      g_autofree gchar *temp_content = read_hwmon_file (temp, &error);
+
+      if (error != NULL)
+          break;
+
+      g_autofree gchar *label = g_strdup_printf ("/sys/class/hwmon/hwmon%d/temp%d_label", hwmonID, i);
+      g_autofree gchar *label_content = read_hwmon_file (label, &error);
+
+      if (error != NULL)
+        {
+          // labels aren't used on ARM
+          g_clear_error (&error);
+          label_content = g_strdup_printf ("Core %d", i);
+        }
+
+      g_strchomp (label_content);
+
+      // ignore Tctl on AMD devices
+      if (!g_str_equal (label_content, "Tctl"))
+        {
+          guint32 temperature;
+          g_autofree gchar *instance = g_strdup_printf ("hwmon%d %s", hwmonID, label_content);
+          sscanf (temp_content, "%"G_GUINT32_FORMAT, &temperature);
```

Comment:
I guess this is also more usually written like `g_ascii_stroull()` or friends...